src/doc/design_overview/overview.rst

```diff
@@ -0,0 +1,961 @@
+in their web browser, the HAproxy configuration serves it the ``index.html``
+from its default backend, which happens to be a simple nginx server dedicated
+to static files.  It also gets images, and the client JavaScript from the
+static server.  Once the JavaScript takes over everything else happens in the
+browser including setting up the appropriate view for the client (whether or
+not they're logged in, etc.) and communicating with the hub using
+Primus/Engine.IO (through the HAproxy--more on that next).  The majority of the
+client is implemented in the code in ``smc-webapp`` and ``webapp-lib``, with
+some bits from ``smc-util``.
+
+HAproxy
+^^^^^^^
+
+HAproxy serves as the front line to all connections from clients to SMC.  It
+routes all connections to different backends depending, primarily, on the
+URL (and port).  The main frontend is of course HTTPS over port 443.  By
+default requests are sent to the static file server (nginx) as mentioned
+above.  Most other requests are sent to the 'hub' backend, which may be
+running any number of the hub servers, one of which is selected using the
+currently configured load balancing scheme (it also uses a session cookie to
+keep individual clients connected to the same hub instance).
+
+Hub
+^^^
+
+The "Hub" is the primary server backend for SMC, built on Node.js as described
+previously.  It consists of an HTTP server with Primus + Engine.IO attached to
+handle real-time bidirectional client/hub communication.  Most communication
+between the Client and the backend happens through the Hub, whose HTTP server
+uses Express to route requests to different services (account management,
+project management, payment, etc.).  Each Hub also sets up a
+``ComputeServerClient`` which gives it access to all the running compute
+servers (discussed next).  The names and URLs of all the available compute
+servers live in a system table in the database.
+
+It also uses `node-http-proxy`_ to create an HTTP proxy server associated
+with each Hub (on port number one higher than the Hub's HTTP port).  If we
+understand correctly, the proxy handles all requests that are to be
+forwarded to individual compute nodes (such as requesting files, or
+resources on web servers belonging to a specific project).  HAproxy doesn't
+know anything about the compute nodes themseves--it just sees URLs that look
+like they belong to a project (they begin with a project UUID) and forwards
+those requests to the Hub's proxy, which in turn checks that the requester
+is authenticated and has permissions to access that project's resources.
+The proxy then forwards the request to the appropriate port on that
+project's compute node::
+
+    Client <--> HAProxy <--> Hub Proxy <--> Compute
+
+Compute
+^^^^^^^
+
+Compute servers are where the real work gets done in SMC projects.  Every
+project is associated with a specific compute server where all their data
+is stored (by way of storage servers mounted on the compute node) and where
```

Comment:
I see.  A couple weeks ago @fcayre and I were poking around in an SMC project and were confused, because we figured `/projects` would be mounted on some networked volume, though it wasn't clear that was the case (though I assume it is on some level within Google's SAN).  

Are projects ever moved to a different compute node, and if so how?